app/lib/search/index_simple.dart

```diff
@@ -33,6 +33,21 @@ class SimplePackageIndex implements PackageIndex {
   bool get isReady => _isReady;
 
   @override
+  Future<bool> containsPackage(String package,
+      {String version, Duration maxAge}) async {
+    final String url = pubUrlOfPackage(package);
+    final PackageDocument doc = _documents[url];
+    if (doc == null) return false;
+    if (version != null && doc.version != version) return false;
```

Comment:
Since `_documents[url]` seems to return only a single `PackageDocument` I assume if this method is not called with the latest version, it will therefore always return false. Is that the case? Does the caller guarantee it's always calling with the latest `{version: ...}`?

Reply:
I wouldn't call it guarantee, but yeah, the notification client only passes the package name and no version, while the task source will get only the latest versions.